p2p/pex/pex_reactor.go

```diff
@@ -543,6 +543,12 @@ func (r *Reactor) dialAttemptsInfo(addr *p2p.NetAddress) (attempts int, lastDial
 }
 
 func (r *Reactor) dialPeer(addr *p2p.NetAddress) error {
+	if r.Switch.Peers().Size() == 0 {
```

Comment:
We may want to upstream this. Seems inocuous

Reply:
We added this in our fork for logging peers for a node (see this [PR](https://github.com/maticnetwork/tendermint/pull/32)). I guess we'll have to find an alternate way to monitor the same. I'll also let @pratikspatil024 chime in here.

Reply:
It's in our TODO list of changes pending to upstream

Reply:
Resolved